modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -260,18 +260,46 @@ using godot_plugins_initialize_fn = bool (*)(void *, GDMonoCache::ManagedCallbac
 #endif
 
 #ifdef TOOLS_ENABLED
+String get_project_assembly_path(const String &p_extension) {
+	String assembly_name = GLOBAL_GET("dotnet/project/assembly_name");
+
+	if (assembly_name.is_empty()) {
+		assembly_name = ProjectSettings::get_singleton()->get_safe_project_name();
+	}
+
+	String assembly_path = GodotSharpDirs::get_res_temp_assemblies_dir()
+								   .path_join(assembly_name + p_extension);
+	return ProjectSettings::get_singleton()->globalize_path(assembly_path);
+}
+
+void try_load_runtime(load_assembly_and_get_function_pointer_fn &r_load_assembly_and_get_function_pointer, const String &p_config_path) {
+	if (r_load_assembly_and_get_function_pointer != nullptr) {
+		return;
+	}
+	if (!FileAccess::exists(p_config_path)) {
+		return;
+	}
+	r_load_assembly_and_get_function_pointer = initialize_hostfxr_for_config(get_data(str_to_hostfxr(p_config_path)));
+}
+
 godot_plugins_initialize_fn initialize_hostfxr_and_godot_plugins(bool &r_runtime_initialized) {
 	godot_plugins_initialize_fn godot_plugins_initialize = nullptr;
 
 	HostFxrCharString godot_plugins_path = str_to_hostfxr(
 			GodotSharpDirs::get_api_assemblies_dir().path_join("GodotPlugins.dll"));
 
-	HostFxrCharString config_path = str_to_hostfxr(
-			GodotSharpDirs::get_api_assemblies_dir().path_join("GodotPlugins.runtimeconfig.json"));
-
-	load_assembly_and_get_function_pointer_fn load_assembly_and_get_function_pointer =
-			initialize_hostfxr_for_config(get_data(config_path));
-	ERR_FAIL_NULL_V(load_assembly_and_get_function_pointer, nullptr);
+	load_assembly_and_get_function_pointer_fn load_assembly_and_get_function_pointer = nullptr;
+
+	// first, try to load the runtime specified by the project
+	try_load_runtime(load_assembly_and_get_function_pointer, get_project_assembly_path(".runtimeconfig.json"));
+	// fall back to that latest installed runtime
+	try_load_runtime(load_assembly_and_get_function_pointer, GodotSharpDirs::get_api_assemblies_dir().path_join("GodotPlugins.runtimeconfig.json"));
+	if (load_assembly_and_get_function_pointer == nullptr) {
+		// show a message box to the user, because the editor will crash before any other messages can be shown
+		OS::get_singleton()->alert(TTR("Unable to load .NET runtime.\n\nPlease install the .NET SDK 6.0 or later from https://dotnet.microsoft.com/en-us/download or use a standard build of Godot."), TTR("Failed to load .NET runtime"));
```

Comment:
I'm not sure we should recommend using a standard build, it's likely that if the user run into this alert they have a C# project and using a standard build wouldn't resolve the issue for them.